src/main/java/dev/codesupport/web/common/configuration/JwtConfiguration.java

```diff
@@ -0,0 +1,75 @@
+package dev.codesupport.web.common.configuration;
+
+import dev.codesupport.web.common.exception.ConfigurationException;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Configurations for JWTs
+ * <p>This is populated automatically at startup by spring via the application.yml files.</p>
+ */
+@Data
+@Configuration
+// This sets the class to use values from the application.yml files.
+@ConfigurationProperties(prefix = "security.jwt")
+public class JwtConfiguration {
+
+    /**
+     * The acceptable value pattern for expiration time expressions, ex:
+     * 5s
+     * 5m
+     * 5h
+     */
+    private static final Pattern timePattern = Pattern.compile("^\\d+[msh]?$");
+    /**
+     * Unit -> coefficient map for converting time expressions to millisecond values
+     */
+    private static final Map<String, Integer> coefficients;
```

Comment:
Using a switch statement would be much higher complexity than a simple map lookup, that being said, I'd prefer to do something other then a map too but I couldn't think of a simpler method at the time.

Reply:
https://docs.oracle.com/javase/8/docs/technotes/guides/language/strings-switch.html

Christ you weren't joking, it does linear comparisons on strings instead of generating a hash?

I can't think of a simpler way around this either given the options Java provides out of the box.